src/node-red/getting-started/building-node-red-plugin.md

```diff
@@ -0,0 +1,309 @@
+---
+eleventyNavigation:
+  key: Node-RED Plugin
+  order: 3
+  parent: Getting Started
+meta:
+  title: Building Node-RED Plugin
+  description: Learn what plugins are and how to build a Node-RED plugin from scratch with our comprehensive step-by-step guide.
+  keywords: node-red plugin, how to build node-red plugin, building node-red plugin
+---
+
+# {{ meta.title }}
+
+This guide aims to provide you with an understanding of what Node-RED plugins are and the differences between nodes and plugins. Additionally, it will walk you through the process of creating a Node-RED plugin from scratch. In this practical guide, we will develop a simple plugin that displays deployment information, including the total deployment count and the timestamp of the last deployment, in a newly added sidebar.
+
+## What Node-RED Plugins Are?
+
+Node-RED plugins are powerful npm modules designed to extend the functionality of the Node-RED editor and runtime. They enable developers to customize and enhance the Node-RED environment by adding sidebar features, modifying existing ones, or integrating with various tools and services. These plugins can create custom panels or offer additional configuration options tailored to user needs.
+
+For example, imagine a plugin introducing a new sidebar to the Node-RED editor to monitor system performance. This plugin could provide real-time insights into CPU and memory usage, helping developers identify bottlenecks and optimize their flows effectively.
+
+## What's the Difference Between Nodes and Plugins?
+
+Many people confuse nodes and plugins, often considering them the same. However, they serve distinct roles. Nodes are the fundamental components used to build flows, with each node performing a specific function, such as data input, processing, or output. They are the building blocks that define the logic and behavior of a flow by connecting various nodes. In contrast, plugins extend the capabilities of the Node-RED editor and runtime. They allow for customization and enhancement of the environment by adding new sidebars with features. Essentially, plugins do not introduce new nodes to your Node-RED palette.
+
+## Building a Node-RED Plugin
+
+This section will create a simple Node-RED plugin that adds a new sidebar panel to the Node-RED editor. This example will guide you through setting up the required files and coding the functionality step by step.
+
+### Step 1: Create the Plugin Directory
+
+First, let’s create a directory for your Node-RED plugin. This keeps everything organized. Open your terminal and run the following commands:
+
+```bash
+mkdir node-red-plugin-deploy-count
+cd node-red-plugin-deploy-count
+```
+
+This creates a folder called `node-red-plugin-deploy-count` and navigates into it. Now, you have a dedicated space for your plugin files.
```

Comment:
Creating a new directory should be done without a terminal I think, to keep it more accessible. What do you think?

Reply:
Same goes for the `touch` usage below. I understand it shows you know how to do this, and it's cool at some level, but how do we get 100.000 more people developing on Node-RED? I think it requires a lot of documentation that's accessible to all.